manifests/init.pp

```diff
@@ -11,6 +11,8 @@
   Pattern[/\d+ KB/] $maximum_object_size_in_memory = $squid::params::maximum_object_size_in_memory,
   String            $package_name                  = $squid::params::package_name,
   String            $service_name                  = $squid::params::service_name,
+  Optional[String]  $error_directory               = $squid::params::error_directory,
```

Comment:
Since you've made these parameters `Optional` (which is a good idea), you should almost certainly default them to `undef`.